<launch>
    <arg name="env_path" default="/home/ubuntu/S13P21A507/sim/simulator2/catkin_ws/.env" />

    <group ns="police_car">
        <node pkg="ssafy_1" type="vehicle_registration.py" name="registration" output="screen">
            <param name="car_name" value="POLICE_CAR" />
            <param name="vehicle_type" value="0" /> <param name="env_path" value="$(arg env_path)" />
            <remap from="/Ego_topic" to="/Ego_topic" />
        </node>

        <node pkg="ssafy_1" type="vehicle_locate_info.py" name="location_updater" output="screen" respawn="true">
             <param name="env_path" value="$(arg env_path)" />
             <remap from="/Ego_topic" to="/Ego_topic" />
        </node>
        
        <node pkg="ssafy_1" type="vehicle_stat_info2.py" name="status_sender" output="screen" respawn="true">
            <param name="env_path" value="$(arg env_path)" />
            <remap from="/Ego_topic" to="/Ego_topic" />
            <remap from="/VehicleCollisionData" to="/VehicleCollisionData" />
        </node>

        <node pkg="ssafy_1" type="LaneFollowingPursuitNode2.py" name="pursuit_controller" output="screen">
            <remap from="/Ego_topic" to="/Ego_topic" />
            <remap from="/ctrl_cmd" to="/ctrl_cmd" />
            <remap from="/Ego_2/Ego_topic" to="/Ego_2/Ego_topic" />
        </node>

        <node pkg="ssafy_1" type="chase_manager.py" name="chase_manager" output="screen" respawn="true">
            <param name="env_path" value="$(arg env_path)" />
            <remap from="/Ego_topic" to="/Ego_topic" />
            <remap from="/ctrl_cmd" to="/ctrl_cmd" />
            <remap from="/fugitive_collision" to="/Ego_2/VehicleCollisionData" />
        </node>
    </group>

    <group ns="fugitive_car">
        <node pkg="ssafy_1" type="vehicle_registration.py" name="registration" output="screen">
            <param name="car_name" value="FUGITIVE" />
            <param name="vehicle_type" value="1" /> <param name="env_path" value="$(arg env_path)" />
            <remap from="/Ego_topic" to="/Ego_2/Ego_topic" />
        </node>

        <node pkg="ssafy_1" type="vehicle_locate_info.py" name="location_updater" output="screen" respawn="true">
             <param name="env_path" value="$(arg env_path)" />
             <remap from="/Ego_topic" to="/Ego_2/Ego_topic" />
        </node>

        <node pkg="ssafy_1" type="vehicle_stat_info2.py" name="status_sender" output="screen" respawn="true">
            <param name="env_path" value="$(arg env_path)" />
            <remap from="/Ego_topic" to="/Ego_2/Ego_topic" />
            <remap from="/VehicleCollisionData" to="/Ego_2/VehicleCollisionData" />
        </node>
    </group>

</launch>
